lib/postinstall.js

```diff
@@ -2,12 +2,22 @@
 
 const fs = require('fs');
 const path = require('path');
+const { promisify } = require('util');
+
+const writeFileAsync = promisify(fs.writeFile);
+const readFileAsync = promisify(fs.readFile);
+
 // get the path to the host project.
 // process.env.PWD doesn't resolve symlinks, e.g. when developing using npm link
 const projectPath = path.resolve(process.env.PWD, '..', '..');
+console.log('Checking configuration of ', projectPath);
 
-const writeEslintRc = (new Promise((resolve, reject) => {
+/**
+ * Writes .eslintrc.js if it doesn't exist. Warns if it exists.
+ */
+const writeEslintRc = async () => {
```

Comment:
Doesn't need to be async, only returns a promise

Reply:
True, but sometimes I feel like it's a good marker to have the async keyword there to mark it as being async, even if it doesn't use await. Shouldn't do any harm.

Reply:
https://github.com/Microsoft/TypeScript/issues/22024#issuecomment-366548045

Reply:
Fine by me, it just wraps the function in a promise.

eg.

```js
async function test() {
  console.log("cat");
}

test().then(() => console.log("kissa"));
```

which might or might not be expected :)